### ISSUE TITLE
Use get_unique_name

### ISSUE THREAD

**Issue by hnandiwada** [timakro/apispec-oneofschema#10]
In the base OpenAPIConverter's resolve_nested_schema function, we see the following lines:
https://github.com/marshmallow-code/apispec/blob/0b2ff4506f88ab2c011b4da0b8c2ccd508c536b1/src/apispec/ext/marshmallow/openapi.py#L134-L135
Before adding a schema to components, the default converter uses the `get_unique_name` function to avoid hitting a `DuplicateComponentNameError`. The `OneofOpenAPIConverter` should do the same, so users can name their schemas in the same way they would without using this plugin. 
I believe this is a two-line change:
```
                unique_component_name = common.get_unique_schema_name(
                    self.spec.components, component_name
                )
                self.spec.components.schema(unique_component_name, schema=type_schema)
```

**Comment by timakro** [comment]
Looks good. Can you open a PR?

**Comment by hnandiwada** [comment]
Done! https://github.com/timakro/apispec-oneofschema/pull/11

**Comment by timakro** [comment]
Released 3.0.1 with your change: https://pypi.org/project/apispec-oneofschema/3.0.1/
